lib/setup/profile.js

```diff
@@ -66,7 +69,104 @@ var appendScript = function(name, options) {
 };
 
 /**
- * Update `~/.bash_profile` and `.zshrc` shell profile files.
+ * Chooses the correct shell init script to insert avn initialization string to,
+ * based on the following criteria:
+ * 1. Append to the one that's present, if the other doesn't exist,
+ * 2. If both exist, append to the one that is sourced by the other, or
+ * 3. If both exist, but neither sources the other,
+ *    append to .bashrc if platform is 'linux',
+ *    or append to .bash_profile if platform is 'darwin' (i.e. macOS)
+ *
+ * @private
+ * @function setup.profile.~shellInitScriptChooser
+ * @return {Promise}
+ */
+var shellInitScriptChooser = function() {
+  var bashrcFile = path.join(process.env.HOME, bashrcFileName);
+  var bashProfileFile = path.join(process.env.HOME, bashProfileFileName);
+  var bashProfileNotExist;
+  var bashrcNotExist;
+
+  // Notice: will change fs.stat to fs.access in the future
+  // Reason: NodeJS 0.10 doesn't have fs.access (added in NodeJS 0.11.15)
```

Comment:
Since `avn` always uses the active node version, we'll likely support `0.10` for _a long time_. This comment could be removed.